### ISSUE TITLE
Bug about your IMTL code

### ISSUE THREAD

**Issue by ZZX-11** [JohnLaMaster/Impartial-Multi-Task-Learning#2]
I think your IMTL code have bug.
if follow your code : `parameters = itertools.chain(model.parameters(), IMTL.parameters())`
and run `list(IMTL.parameters())` will get a empty list.  Also, after `optimizer.step()`,  the **s_t** will not update.
![image](https://github.com/JohnLaMaster/Impartial-Multi-Task-Learning/assets/113749872/7ca23983-8eac-453a-a6fa-b1bd32d6d230)

The problem comes from you not initial `self.register_parameter('s_t',nn.Parameter(torch.ones(self.num_losses+1).squeeze(),requires_grad=True))` 
in the `init()` function, therefore `optimizer` will not update **s_t**
I have changed your code like this :

![image](https://github.com/JohnLaMaster/Impartial-Multi-Task-Learning/assets/113749872/5b54dac8-ca14-4fec-b536-4a6f83fde96b)

**Comment by ZZX-11** [comment]
And when I use `gradient` mode, the NN can't get grad, and get updated

**Comment by JohnLaMaster** [comment]
Hi, thank you for pointing that out. I have fixed the code. When defining the module, you will now need to add the expected number of losses as an input. The README documentation was also updated. 

Can you give me some details regarding the issue with the gradients? Was that dues to the initialization error, which sould now be resolved, or was that still an issue after your modification?

**Comment by ZZX-11** [comment]
Hi, I've find an better implementation of IMTL and other multitask algorithms. The MTL methods offered are plug-and-play, making them exceptionally convenient to use:  
1. https://github.com/AvivNavon/nash-mtl/blob/main/methods/weight_methods.py
2. https://github.com/Cranial-XIX/FAMO/blob/main/methods/weight_methods.py
